external/CMakeLists.txt

```diff
@@ -7,12 +7,17 @@
 # silent otherwise
 set(CMAKE_MESSAGE_LOG_LEVEL VERBOSE)
 
-# Download the package and add to a list
+# Download the package, save version, add to a list
 set(_celer_builtin_packages)
-macro(celer_fetch_external name url sha)
+macro(celer_fetch_external name version url sha)
+  # Save version and annotated string for corecel/Config.cc include
+  set(${name}_VERSION ${version} PARENT_SCOPE)
+  set(${name}_VERSION_STRING "${version}+builtin" PARENT_SCOPE)
+
+  string(REGEX REPLACE "VERSION" "${version}" _url "${url}")
```

Comment:
There might be a better way to do this... I couldn't think of a suitable substitution expression which made it clear that the function would update the URL as part of a template.

Reply:
Oh I like that. On it!